orderbook/src/api.rs

```diff
@@ -70,6 +68,27 @@ pub fn handle_all_routes(
     routes_with_labels.recover(handle_rejection).with(cors)
 }
 
+#[derive(Clone)]
+pub struct OrderQuoter {
+    fee_calculator: Arc<dyn MinFeeCalculating>,
+    price_estimator: Arc<dyn PriceEstimating>,
+    order_validator: Arc<dyn OrderValidating>,
+}
+
+impl OrderQuoter {
+    pub fn new(
+        fee_calculator: Arc<dyn MinFeeCalculating>,
+        price_estimator: Arc<dyn PriceEstimating>,
+        order_validator: Arc<dyn OrderValidating>,
+    ) -> Self {
+        Self {
+            fee_calculator,
+            price_estimator,
+            order_validator,
+        }
+    }
+}
+
```

Comment:
Why not put this type into the `post_quote` module seeing as that is where it gets used. This way the type is closer to the logic that uses it.

Reply:
I feel you can make all of the pub and not need pub(crate).